core/src/modules/ops/record.ts

```diff
@@ -969,11 +987,21 @@ export namespace RecordOps {
       }
     );
 
-    // For postgres only: we can update our result set with the rows that were updated, filtering out those which are no longer state=pending
-    // in SQLite this isn't possible, but contention is far less likely
-    if (updateResponse[1]) records = updateResponse[1];
-
-    if (records.length === 0) return [];
+    // Update records to invalid if any assocaited properties are invalid.
+    await api.sequelize.query(`
+      UPDATE
+        "records"
+      SET
+        "invalid" = TRUE
+      WHERE
+        "records"."id" IN(
+          SELECT
+            "recordProperties"."recordId" FROM "recordProperties"
+          WHERE
+            "recordProperties"."recordId" IN(${records.map((p) => `'${p.id}'`)})
+            AND "recordProperties"."invalidReason" IS NOT NULL)
+        AND "records"."invalid" = FALSE;
```

Comment:
I thought I had this working without a sub-query but I was wrong. Not including a sub-query just resulted in all matching records getting set to invalid if _any_ were invalid.